dlt/destinations/impl/filesystem/filesystem.py

```diff
@@ -894,14 +896,17 @@ def get_open_table_catalog(self, table_format: TTableFormat, catalog_name: str =
 
         return catalog
 
-    def get_open_table_location(self, table_format: TTableFormat, table_name: str) -> str:
+    def get_open_table_location(
+        self, table_format: TTableFormat, table_name: str
+    ) -> Tuple[str, bool]:
         """All tables have location, also those in "native" table format."""
         folder = self.get_table_dir(table_name)
+        prefix = self.get_table_prefix(table_name)
         location = self.make_remote_url(folder)
         if self.config.is_local_filesystem and os.name == "nt":
             # pyiceberg cannot deal with windows absolute urls
             location = location.replace("file:///", "file://")
-        return location
+        return location, (folder == prefix or (folder + os.path.sep) == prefix)
```

Comment:
maybe `make_remote_url` removes trailing separator from `folder`?

but overall `filesystem` does not enforce the convention that folder should always end with separator. maybe we should unit test a few functions there to make sure they follow this convention

Reply:
I am ensuring now that this function returns a url with a trailing slash if it is a folder (it has to be removed in some places downstream again for things to work)